packages/button/src/Button.tsx

```diff
@@ -99,31 +101,15 @@ const Button: ExtendableBox<
     showFocus,
   });
 
-  const isAnchor = typeof rest.href === 'string';
-  let type: JSX.IntrinsicElements['button']['type'];
-
-  // Expecting this error based on typing in Box component
-  // @ts-expect-error rest.as may be defined
-  if ((rest.as && rest.as === 'button') || (!isAnchor && !rest.as)) {
-    type = 'button';
-  }
-
-  // Render a disabled link as a button to retain focusability
-  const getButtonTag = (
-    isAnchor: boolean,
-    disabled: boolean,
-  ): 'a' | 'button' => {
-    if (isAnchor && !disabled) return 'a';
-    return 'button';
-  };
+  const isAnchor: boolean = (!!rest.href || as === 'a') && !disabled;
 
   const buttonProps = {
-    type,
+    type: type || 'button',
```

Comment:
I think type should still be `undefined` if it's an anchor

Reply:
I defaulted the value here to `'button'` because of what used to be on line 107. Should we change this?

Reply:
`type` would still be undefined there if `as=='a'` or if `href` is defined

Reply:
My understanding here is that this prop allows devs to set the [button type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type) (e.g. submit, reset) on a LG button